### PR TITLE
Backported BackupScheduled fixes from skpr/operator.

### PR DESCRIPTION
**Pull request by nicksantamaria**

### CODE REVIEW

File: pkg/controller/backupscheduled/backupscheduled_controller.go [hunk 1]
```diff
@@ -81,95 +104,244 @@ func (r *ReconcileBackupScheduled) Reconcile(request reconcile.Request) (reconci
...
+
+	if scheduled.Spec.Schedule.Suspend != nil && *scheduled.Spec.Schedule.Suspend {
+		log.Info("Scheduling has been suspended. Skipping.")
+		return reconcile.Result{}, nil
+	}
+
+	return r.ScheduleNextBackup(scheduled, active)
+}
+
+func getScheduleComparison(s shpmetav1.ScheduledStatus, now time.Time) time.Time {
```
Comment (acbramley) [inline pkg/controller/backupscheduled/backupscheduled_controller.go]: Doesn't look like this is used anywhere?

File: pkg/utils/k8s/sync/certificate.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	awsv1beta1 "github.com/skpr/operator/pkg/apis/aws/v1beta1"
+)
+
+// Certificate function to ensure the Object is in sync.
+func Certificate(parent metav1.Object, spec awsv1beta1.CertificateSpec, scheme *runtime.Scheme) controllerutil.MutateFn {
```
Comment (acbramley) [inline pkg/utils/k8s/sync/certificate.go]: Lots of files in this sync folder that we don't need?

File: pkg/utils/k8s/deployment/deployment.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package deployment
```
Comment (acbramley) [inline pkg/utils/k8s/deployment/deployment.go]: What's this used for?

Comment (nicksantamaria) [reply pkg/utils/k8s/deployment/deployment.go]: Likely nothing - will prune it.

File: Makefile [hunk 4]
```diff
@@ -35,7 +30,10 @@ install: manifests
 kustomize:
 	@echo "updating kustomize namespace to ${NAMESPACE}"
 	sed -i'' -e 's@namespace: .*@namespace: '"${NAMESPACE}"'@' ./config/default/kustomization.yaml
-	kustomize build config/default -o ./config/deploy.yaml
```
Comment (nicksantamaria) [inline Makefile:38]: Moving the `kustomize` commands into docker are necessary due to the older version required by kubebuilder v1.

Comment (pingers) [reply Makefile:38]: Which we need for our production hosting version right...

File: pkg/utils/scheduled/utils.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/robfig/cron"
+
+	shpmetav1 "github.com/universityofadelaide/shepherd-operator/pkg/apis/meta/v1"
+)
+
+// GetScheduledTime from an annotation map.
+func GetScheduledTime(annotations map[string]string) (*time.Time, error) {
```
Comment (pingers) [inline pkg/utils/scheduled/utils.go:13]: I'm guessing these were ripped out of one of the deleted deps?

**Comment by pingers** [comment]
I don't time to build it at the moment, but looks pretty good to me.

File: pkg/controller/backupscheduled/backupscheduled_controller.go (hunk 1 above)
Comment (nicksantamaria) [reply pkg/controller/backupscheduled/backupscheduled_controller.go]: Removed! thanks spotting it 👍

File: pkg/utils/scheduled/utils.go (hunk 5 above)
Comment (nicksantamaria) [reply pkg/utils/scheduled/utils.go:13]: Correct - everything in these `utils` directory is from skpr/operator (RIP).

**Pull request merged**